std/getopt.d

```diff
@@ -611,6 +612,13 @@ private template optionValidator(A...)
     static assert(optionValidator!(S,S,S,P) != "");
 }
 
+unittest
+{
+    bool opt;
+    string[] args = ["program", "-a"];
+    static assert( is(typeof(getopt(args, config.passThrough, 'a', &opt))));
```

Comment:
Can you change this to 
```
getopt(args, config.passThrough, 'a', &opt)
assert(opt);
```
(or something similar), so that the run time code is also verifed, in addition to the template constraints?

Otherwise, LGTM.